### ISSUE TITLE
[Bug] AttributeError: 'Namespace' object has no attribute 'silent'

### ISSUE THREAD

**Issue by nerdneilsfield** [redromnon/HeroicBashLauncher#127]
**Describe the bug**

can not run the HeroicBash Luancher,  `AttributeError: 'Namespace' object has no attribute 'silent'`

**Expected behavior**

the script should run and add the games to Steam Library

**Screenshots**

log

```
(deck@steamdeck HeroicBashLauncher)$ ./HeroicBashLauncher 
Using Bash Launcher v3.2.2
NOTE - This is an independent project and not affiliated with Heroic Games Launcher.

3.32.0
Traceback (most recent call last):
  File "func/HeroicBashLauncher.py", line 69, in <module>
  File "func/listinstalled.py", line 50, in listinstalled
  File "func/createlaunchfile.py", line 17, in createlaunchfile
AttributeError: 'Namespace' object has no attribute 'silent'
[1321720] Failed to execute script 'HeroicBashLauncher' due to unhandled exception!

```

**Details**

 - Platform: Steam Deck
 - OS: Steam OS 3
 - Heroic Version: v2.7.1 flatpak
 - Bash Launcher Version: V3.2.2

**Additional context**
None

**Comment by redromnon** [comment]
This issue is the same as #125 
Looking into it.

**Comment by nerdneilsfield** [comment]
Thanks

**Comment by redromnon** [comment]
Hey @afwolfe, do you know a solution to this? I've not encountered it on my side but several users are complaining.
Linked a [commit](https://github.com/redromnon/HeroicBashLauncher/commit/066199bda0d9c0e23ea17bfea28434fe0180dcc1) I made to _kind of_ refactor it.

**Comment by redromnon** [comment]
@nerdneilsfield Can you check if [this build](https://github.com/redromnon/HeroicBashLauncher/actions/runs/5066680911) works? Just replace the executable with the one in the Bash Launcher folder.

**Comment by afwolfe** [comment]
Hey @redromnon. It took some digging and trial and error but I think I was able to figure it out.

In order to debug with VS Code, I had to do the following to get HeroicBashLauncher to work without being bundled up by PyInstaller and being in the Games/Heroic/HeroicBashLauncher directory (please let me know if there's a better way to do this):
1. Copying HeroicBashLauncher.py to the root of the repo (because it didn't like the `func` imports while it was in the same directory.)
2. Using this launch.json: (replacing `${REPLACE_WITH_PATH_TO_HOME}`)
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "HeroicBashLauncher",
            "type": "python",
            "request": "launch",
            "program": "${workspaceFolder}/HeroicBashLauncher.py",
            "cwd": "${REPLACE_WITH_PATH_TO_HOME}/Games/Heroic/HeroicBashLauncher/",
            "console": "integratedTerminal",
            "justMyCode": true
        }
    ]
}
```
3. I set a breakpoint at createlaunchfile.py:11 and confirmed that `args` is `None` when imported. On the `fix-args-parser` branch `args` is a `Namespace`, but it looks like `args.silent` is still unset and will throw the `AttributeError: 'Namespace' object has no attribute 'silent'`.

After a bit more searching, I think this StackOverflow does a good job explaining the bug which seems to relate to the visibility of global variables in imported modules: https://stackoverflow.com/questions/15959534/visibility-of-global-variables-in-imported-modules
Especially this in the top answer from abarnert:
> Don't use a `from` import unless the variable is intended to be a constant. `from shared_stuff import a` would create a new a variable initialized to whatever `shared_stuff.a` referred to at the time of the import, and this new `a` variable would not be affected by assignments to `shared_stuff.a`.

TL;DR: I think one solution could be to change createlaunchfile.py and anywhere else that's using `from func.settings import args` to `from func import settings` and changing any references from `args` to `settings.args` should resolve it and use the module's instance of the variable (with the parsed arguments) as we'd expect.

**Comment by redromnon** [comment]
Thanks a ton @afwolfe 
Will look at the probable solution for this. 
Did you also find out why this issue doesn't affect all users?

**Comment by redromnon** [comment]
HeroicBashLauncher must have contained it under `HeroicBashLauncher.log`. I'll make it a must to add the log file too.

**Comment by redromnon** [comment]
Made the changes and hopefully, it works fine. This is the [updated build](https://github.com/redromnon/HeroicBashLauncher/actions/runs/5080442448).
